### PR TITLE
refactor!: remove gateway token support

### PR DESCRIPTION
**Pull request by milapsheth**
[AXE-6239]

[AXE-6239]: https://axelarnetwork.atlassian.net/browse/AXE-6239?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/TokenHandler.sol [hunk 1]
```diff
@@ -85,15 +70,9 @@ contract TokenHandler is ITokenHandler, ITokenManagerType, ReentrancyGuard, Crea
...
-     * @return symbol The symbol for the token, if not empty the token is a gateway token and a callContractWith token has to be made.
      */
     // slither-disable-next-line locked-ether
-    function takeToken(
-        bytes32 tokenId,
-        bool tokenOnly,
-        address from,
-        uint256 amount
-    ) external payable returns (uint256, string memory symbol) {
+    function takeToken(bytes32 tokenId, bool tokenOnly, address from, uint256 amount) external payable returns (uint256) {
```
Comment (ahramy) [inline contracts/TokenHandler.sol:75]:
```suggestion
    function takeToken(bytes32 tokenId, bool tokenOnly, address from, uint256 amount) external payable returns (uint256 amount) {
```

File: contracts/TokenHandler.sol [hunk 2]
```diff
@@ -109,17 +88,14 @@ contract TokenHandler is ITokenHandler, ITokenManagerType, ReentrancyGuard, Crea
...
-            _transferTokenFrom(tokenAddress, from, address(this), amount);
         } else {
             revert UnsupportedTokenManagerType(tokenManagerType);
         }
 
         /// @dev Track the flow amount being sent out as a message
         ITokenManager(tokenManager).addFlowOut(amount);
 
-        return (amount, symbol);
+        return amount;
```
Comment (ahramy) [inline contracts/TokenHandler.sol:98]: we can remove `return amount;` with named return

File: test/InterchainTokenService.js [hunk 3]
```diff
@@ -2667,15 +2333,13 @@ describe('Interchain Token Service', () => {
...
         let tokenId;
         let executable;
-        let invalidExecutable;
         let token;
 
         before(async () => {
-            [token, , tokenId] = await deployFunctions.gateway(service, tokenName, tokenSymbol, tokenDecimals, amount * 2, true);
-            await token.approve(service.address, amount * 2).then((tx) => tx.wait);
+            [token, , tokenId] = await deployFunctions.mintBurn(service, tokenName, tokenSymbol, tokenDecimals, amount * 2, true);
+            // await token.approve(service.address, amount * 2).then((tx) => tx.wait);
```
Comment (ahramy) [inline test/InterchainTokenService.js:2340]: unnecessary comment, please remove it

File: test/InterchainTokenService.js [hunk 4]
```diff
@@ -1510,86 +1433,12 @@ describe('Interchain Token Service', () => {
...
-                '0x',
-                tokenSymbol,
-                amount,
-            );
-
+    describe('Execute with token', () => {
+        it('should revert on execute with token', async () => {
             await expectRevert(
-                (gasOptions) => service.executeWithToken(commandId, sourceChain, wallet.address, '0x', tokenSymbol, amount, gasOptions),
+                (gasOptions) => service.executeWithToken(HashZero, 'sourceChain', wallet.address, '0x', 'TEST', 1, gasOptions),
```
Comment (ahramy) [inline test/InterchainTokenService.js:1439]: Does `commandId` param needs to be changed to `HashZero`?
```suggestion
                (gasOptions) => service.executeWithToken(commandId, 'sourceChain', wallet.address, '0x', 'TEST', 1, gasOptions),
```

File: contracts/InterchainTokenService.sol [hunk 5]
```diff
@@ -756,17 +701,14 @@ contract InterchainTokenService is
...
+        address recipient = (expressExecutor == address(0)) ? destinationAddress : expressExecutor;
+        address tokenAddress;
+        (amount, tokenAddress) = _giveToken(tokenId, recipient, amount);
+        // Return token to the express executor
         if (expressExecutor != address(0)) {
-            // slither-disable-next-line unused-return
-            _giveToken(tokenId, expressExecutor, amount);
-
             return;
         }
```
Comment (ahramy) [inline contracts/InterchainTokenService.sol:711]: How about this for better code readability?
```suggestion
        // Return token to the express executor
        if (expressExecutor != address(0)) {
            _giveToken(tokenId, expressExecutor, amount);
            return;
        }

        address tokenAddress;
        (amount, tokenAddress) = _giveToken(tokenId, destinationAddress, amount);
```

File: contracts/TokenHandler.sol [hunk 6]
```diff
@@ -162,16 +137,10 @@ contract TokenHandler is ITokenHandler, ITokenManagerType, ReentrancyGuard, Crea
...
-
-        // Approve the gateway here. One-time infinite approval works for gateway wrapped tokens, and for most origin tokens.
-        // Approval can be refreshed in the future if needed for certain tokens via an upgrade, but realistically should never be exhausted.
-        if (tokenManagerType == uint256(TokenManagerType.GATEWAY)) {
-            address token = ITokenManager(tokenManager).tokenAddress();
-            _approveGateway(token, UINT256_MAX);
-        }
     }
 
     function _transferTokenFrom(address tokenAddress, address from, address to, uint256 amount) internal {
```
Comment (ahramy) [inline contracts/TokenHandler.sol:146]: Upon checking code coverage, the `_transferToken` function is no longer being used.
```
function _transferToken(address tokenAddress, address to, uint256 amount) internal {
    // slither-disable-next-line arbitrary-send-erc20
    IERC20(tokenAddress).safeTransfer(to, amount);
}
```

File: test/InterchainTokenService.js (hunk 4 above)
Comment (milapsheth) [reply test/InterchainTokenService.js:1439]: Yes this is just checking that this method reverts

File: contracts/InterchainTokenService.sol (hunk 5 above)
Comment (milapsheth) [reply contracts/InterchainTokenService.sol:711]: it's a small optimization. `_giveToken` can be inlined just once, but doesn't seem to have a major impact, so I reverted

**Review by ahramy — APPROVED** [review]

**Pull request merged**